packages/ckeditor5-link/src/linkediting.js

```diff
@@ -132,7 +131,10 @@ export default class LinkEditing extends Plugin {
 	 */
 	_enableAutomaticDecorators( automaticDecoratorDefinitions ) {
 		const editor = this.editor;
-		const automaticDecorators = new AutomaticDecorators();
+		// Store automatic decorators in the command instance as we do the same with manual decorators.
+		// Thanks to that, `LinkImageEditing` plugin can re-use the same definitions.
+		const command = editor.commands.get( 'link' );
+		const automaticDecorators = command.automaticDecorators;
```

Comment:
I am fine with introducing the change you described but it will introduce MAJOR BC in `LinkCommand` class because we will expect the second argument in the `LinkCommand` constructor.

Reply:
The whole code would benefit from a refactor but it's OK to keep it the way we already have it.